Framework/API/src/SpectrumInfo.cpp

```diff
@@ -0,0 +1,42 @@
+#include "MantidAPI/GeometryInfoFactory.h"
+#include "MantidAPI/GeometryInfo.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidKernel/MultiThreaded.h"
+#include "MantidKernel/make_unique.h"
+
+namespace Mantid {
+namespace API {
+
+SpectrumInfo::SpectrumInfo(const MatrixWorkspace &workspace)
+    : m_factory{Kernel::make_unique<GeometryInfoFactory>(workspace)},
```

Comment:
I think what's been doing here is good, but taking a look at `GeometryInfo` and how that interplays with `GeometryInfoFactory` I think things are a little confused. `GeometryInfo` is created with a back-reference to the `GeometryInfoFactory` and `GeometryInfoFactory` does a lot more than just create `GeometryInfo`. I think a better design would be for `GeometryInfo` not to keep the back-reference and for `GeometryInfoFactory` to pass all the required parameters (source pos, sample pos etc) as constructor arguments. Or if performance is important some  kind of `InstrumentGeometryCore` containing that information that can be passed by shared_ptr to the `GeometryInfos`